stores/metrics.go

```diff
@@ -66,3 +80,55 @@ func (dbContractMetric) TableName() string         { return "contracts" }
 func (dbContractSetMetric) TableName() string      { return "contract_sets" }
 func (dbContractSetChurnMetric) TableName() string { return "contract_sets_churn" }
 func (dbPerformanceMetric) TableName() string      { return "performance" }
+
+func scopeTimeRange(tx *gorm.DB, after, before *time.Time) *gorm.DB {
+	if after != nil {
+		tx = tx.Where("time > ?", unixTimeMS(*after))
+	}
+	if before != nil {
+		tx = tx.Where("time <= ?", unixTimeMS(*before))
+	}
+	return tx
+}
+
+func (s *SQLStore) contractSetMetrics(ctx context.Context, opts api.ContractSetMetricsQueryOpts) ([]dbContractSetMetric, error) {
```

Comment:
Hm, are we going to pull down our api options all the way down to the store level? Originally I think we decided to define them for use in our clients but our store interfaces would have the parameters in full. I don't really mind but by doing so we're making it somewhat inconsistent. Internal kitchen though but still.

Reply:
Hm it's not really that I mean to use api options in the backend but rather backend options in the api client. `ContractSetMetrics` are not too bad but other metrics will have tons of fields to filter by and having a struct to combine all filters seems a bit nicer.

Reply:
👍  ok